lib/langchain/tool/database.rb

```diff
@@ -0,0 +1,41 @@
+module Langchain::Tool
+  class Database < Base
+    description <<~DESC
+      Useful for getting the result of a database query.
+
+      The input to this tool should be valid SQL.
+    DESC
+
+    # Establish a database connection
+    # example: 'postgres://user:password@localhost:5432/db_name'
+    def initialize(db_connection_string = "")
+      depends_on "sequel"
+      require "sequel"
+      require "sequel/extensions/schema_dumper"
+
+      @db = if db_connection_string.empty?
```

Comment:
I think this Database Tool should just require the `db_connection_string` and if it's empty or nil or not passed in -- an error should be thrown.

Reply:
I'll try, but Sequel throws an akward error.  Do I catch this and return a custom error?

```
  1) Langchain::Agent::SQLQueryAgent#prompt_template returns a prompt template instance
     Failure/Error: @db = Sequel.connect(db_connection_string)

     NoMethodError:
       undefined method `to_sym' for nil:NilClass
     # /Users/mattlindsey/.rvm/gems/ruby-3.2.1/gems/sequel-5.68.0/lib/sequel/database/connecting.rb:17:in `adapter_class'
     # /Users/mattlindsey/.rvm/gems/ruby-3.2.1/gems/sequel-5.68.0/lib/sequel/database/connecting.rb:36:in `connect'
     # /Users/mattlindsey/.rvm/gems/ruby-3.2.1/gems/sequel-5.68.0/lib/sequel/core.rb:124:in `connect'
     # ./lib/langchain/tool/database.rb:16:in `initialize'
     # ./lib/langchain/agent/sql_query_agent/sql_query_agent.rb:17:in `new'
     # ./lib/langchain/agent/sql_query_agent/sql_query_agent.rb:17:in `initialize'
     # ./spec/langchain/agent/sql_query_agent/sql_query_agent_spec.rb:5:in `new'
     # ./spec/langchain/agent/sql_query_agent/sql_query_agent_spec.rb:5:in `block (2 levels) in <top (required)>'
     # ./spec/langchain/agent/sql_query_agent/sql_query_agent_spec.rb:79:in `block (3 levels) in <top (required)>'

Reply:
In the meantime you can just:
```ruby
raise StandardError, "db_connection_string parameter cannot be blank" if db_connection_string.empty?
```

Reply:
Done.  Basic functionality is working.  Anything else for this PR?